pymongoexplain/commands.py

```diff
@@ -108,6 +121,8 @@ class FindCommand(BaseCommand):
     def __init__(self, collection: Collection,
                  kwargs):
         super().__init__(collection.name)
+        if kwargs["filter"] == {}:
+            self.command_document = {}
```

Comment:
This looks like a bug (or it's masking a bug). Let's remove this code and address the failing test in a different way. Please post the test failure here.

For reference this is how pymongo generates a find command:
https://github.com/mongodb/mongo-python-driver/blob/3.11.0rc0/pymongo/message.py#L181-L217

Reply:
Yeah, I thought that might be a bug because it doesn't really make sense. Removing that doesn't seem to cause any additional test failures.

Reply:
The three different tests that fail are below, they are in the format of \<the command we constructed\>, \<the command that was expected\>
```
SON([('find', 'test_find_allowdiskuse'), ('filter', {})])
{}
```
```
SON([('find', 'test_find_allowdiskuse'), ('allowDiskUse', True), ('filter', {})])
{}
```
```
SON([('find', 'test_find_allowdiskuse'), ('allowDiskUse', False), ('filter', {})])
{}
```

Reply:
Can you post the entire failed test output though? Like this:
```
FAIL: test_keyword_arg_defaults (test.test_client.ClientUnitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/shane/git/mongo-python-driver/test/test_client.py", line 128, in test_keyword_arg_defaults
    self.assertEqual(20.1, pool_opts.connect_timeout)
AssertionError: 20.1 != 20.0
```

Reply:
```
SON([('find', 'test_find_allowdiskuse'), ('filter', {})])
{}

Error
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/julius/pymongoexplain/test/__init__.py", line 442, in wrap
    return f(*args, **kwargs)
  File "/Users/julius/pymongoexplain/test/test_crud_v2.py", line 62, in run_scenario
    self.run_scenario(scenario_def, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 615, in run_scenario
    self.run_test_ops(sessions, collection, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 525, in run_test_ops
    self.run_operations(sessions, collection, test['operations'])
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 431, in run_operations
    result = self.run_operation(sessions, collection, op.copy())
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 374, in run_operation
    self._compare_command_dicts(wrapped_collection.last_cmd_payload,
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 269, in _compare_command_dicts
    self.assertEqual(ours[key], theirs[key])
KeyError: 'find
```

```
SON([('find', 'test_find_allowdiskuse'), ('allowDiskUse', True), ('filter', {})])
{}

Error
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/julius/pymongoexplain/test/__init__.py", line 442, in wrap
    return f(*args, **kwargs)
  File "/Users/julius/pymongoexplain/test/test_crud_v2.py", line 62, in run_scenario
    self.run_scenario(scenario_def, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 615, in run_scenario
    self.run_test_ops(sessions, collection, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 525, in run_test_ops
    self.run_operations(sessions, collection, test['operations'])
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 431, in run_operations
    result = self.run_operation(sessions, collection, op.copy())
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 374, in run_operation
    self._compare_command_dicts(wrapped_collection.last_cmd_payload,
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 269, in _compare_command_dicts
    self.assertEqual(ours[key], theirs[key])
KeyError: 'find'
```
```
SON([('find', 'test_find_allowdiskuse'), ('allowDiskUse', False), ('filter', {})])
{}

Error
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/julius/pymongoexplain/test/__init__.py", line 442, in wrap
    return f(*args, **kwargs)
  File "/Users/julius/pymongoexplain/test/test_crud_v2.py", line 62, in run_scenario
    self.run_scenario(scenario_def, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 615, in run_scenario
    self.run_test_ops(sessions, collection, test)
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 525, in run_test_ops
    self.run_operations(sessions, collection, test['operations'])
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 431, in run_operations
    result = self.run_operation(sessions, collection, op.copy())
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 374, in run_operation
    self._compare_command_dicts(wrapped_collection.last_cmd_payload,
  File "/Users/julius/pymongoexplain/test/utils_spec_runner.py", line 269, in _compare_command_dicts
    self.assertEqual(ours[key], theirs[key])
KeyError: 'find'

```